### ISSUE TITLE
ValueError: Layer count mismatch

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#1725]
When loading `retinanet_resnet50_pascalvoc` preset in TF 2.11, it gives error but works fine in TF 2.12.

```python
import keras_cv
import tensorflow as tf

tf.__version__, keras_cv.__version__
('2.11.0', '0.4.2')

pretrained_model = keras_cv.models.RetinaNet.from_preset(
    "resnet50_imagenet",
    bounding_box_format="xywh",
    num_classes=1
)
# OK
```
```python
pretrained_model = keras_cv.models.RetinaNet.from_preset(
    "retinanet_resnet50_pascalvoc", bounding_box_format="xywh"
)
```
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/tmp/ipykernel_27/3432473294.py in <module>
      1 pretrained_model = keras_cv.models.RetinaNet.from_preset(
----> 2     "retinanet_resnet50_pascalvoc", bounding_box_format="xywh"
      3 )

/opt/conda/lib/python3.7/site-packages/keras_cv/models/task.py in from_preset(calling_cls, *args, **kwargs)
    162 
    163             def from_preset(calling_cls, *args, **kwargs):
--> 164                 return super(cls, calling_cls).from_preset(*args, **kwargs)
    165 
    166             cls.from_preset = classmethod(from_preset)

/opt/conda/lib/python3.7/site-packages/keras_cv/models/task.py in from_preset(cls, preset, load_weights, **kwargs)
    150         )
    151 
--> 152         model.load_weights(weights)
    153         return model
    154 

/opt/conda/lib/python3.7/site-packages/keras/utils/traceback_utils.py in error_handler(*args, **kwargs)
     68             # To get the full stack trace, call:
     69             # `tf.debugging.disable_traceback_filtering()`
---> 70             raise e.with_traceback(filtered_tb) from None
     71         finally:
     72             del filtered_tb

/opt/conda/lib/python3.7/site-packages/keras/saving/legacy/hdf5_format.py in load_weights_from_hdf5_group(f, model)
    811     if len(layer_names) != len(filtered_layers):
    812         raise ValueError(
--> 813             "Layer count mismatch when loading weights from file. "
    814             f"Model expected {len(filtered_layers)} layers, found "
    815             f"{len(layer_names)} saved layers."

ValueError: Layer count mismatch when loading weights from file. Model expected 5 layers, found 0 saved layers.
```

**Comment by innat** [comment]
@LukeWood Have you able to reproduced the issue?

**Comment by LukeWood** [comment]
Let me give it a try, you might have an invalid cache-ing situation going on.

**Comment by LukeWood** [comment]
can you try running `rm -rf ~/.keras` on the affected machine?

**Comment by innat** [comment]
I've just tried to run it on a newly created gcp instance notebook, the error is same. The same issue can be reproduced also in kaggle env (tf 2.11) with gpu accelerator.  But with tpu-vm (tf 2.12) it runs.

**Comment by LukeWood** [comment]
interesting; perhaps a TF 2.11 related issue

**Comment by innat** [comment]
Could be. Though I could run yolo-v8 with it (2.11).

**Comment by LukeWood** [comment]
@jbischof is 2.11 safe?  is it confirmed tied to 2.10?

**Comment by jbischof** [comment]
@LukeWood apologies I had python 3.10 and TF 2.11 when I got the failure. Upgrading to TF 2.12 solved the problem.

This issue is blocking us from running `large` tests on GCB because it uses TF 2.11. You can check yourself with pytest:
```
pytest keras_cv/models/object_detection/retinanet/ --run_extra_large --durations=0
```

**Comment by jbischof** [comment]
> It's possible that this is because the weights are in a file called `.weights.h5` instead of `.h5` -- not 100% sure but it's something to try. We'd just need to rename the files in GCS and update the preset.

I tried this in #1822 but no dice...same error.

**Comment by ianstenbit** [comment]
This seems to have been a save format issue and will be fixed by #1824, where I am also updating the weight presets.

**Comment by ianstenbit** [comment]
This should now be fixed by #1824. Let me know if you're still seeing the issue, but I am now seeing success when I run your repro. Thanks for the issue report!
